packages/configuration-svc/src/infrastructure/mongodb_configset_repo.ts

```diff
@@ -0,0 +1,349 @@
+    private _initialized: boolean = false;
+    private readonly _databaseName: string = "configsets";
+    private readonly _collectionNameGlobalConfigSets: string = "globalConfigsets";
+    private readonly _collectionNameBcConfigSets: string = "bcConfigsets";
+    private _globalConfigSets: GlobalConfigurationSet[] = [];
+    private _bcConfigSets: Map<string, BoundedContextConfigurationSet[]> = new Map<string, BoundedContextConfigurationSet[]>();
+
+
+    constructor(_mongoUri: string, logger: ILogger) {
+        this._logger = logger.createChild(this.constructor.name);
+        this._mongoUri = _mongoUri;
+    }
+
+    async init(): Promise<void> {
+        try {
+            this._mongoClient = await MongoClient.connect(this._mongoUri);
+        } catch (err: any) {
+            this._logger.error(err);
+            this._logger.isWarnEnabled() &&
+                this._logger.warn(
+                    `MongoConfigSetRepo - init failed with error: ${err?.message?.toString()}`
+                );
+            throw err;
+        }
+
+        if (this._mongoClient === null)
+            throw new Error("Couldn't instantiate mongo client");
+
+        const db = this._mongoClient.db(this._databaseName);
+
+        const collections = await db.listCollections().toArray();
+
+        // Check if the GlobalConfigSet collection already exists or create.
+        if (collections.find((col) => col.name === this._collectionNameGlobalConfigSets)) {
+            this._collectionGlobalConfigSets = db.collection(this._collectionNameGlobalConfigSets);
+        } else {
+            this._collectionGlobalConfigSets = await db.createCollection(this._collectionNameGlobalConfigSets);         
+        }
+
+        // Check if the BCConfigSet collection already exists or create.
+        if (collections.find((col) => col.name === this._collectionNameBcConfigSets) ) {
+            this._collectionBcConfigSets = db.collection(this._collectionNameBcConfigSets);
+        } else {
+            this._collectionBcConfigSets = await db.createCollection(this._collectionNameBcConfigSets);
+        }
+
+        await this._loadFromGlobalConfigSets();
+        await this._loadFromBcConfigSets();
+
+        this._initialized = true;
+        this._logger.info("MongoConfigSetRepo - initialized");
+
+    }
+    async destroy(): Promise<void> {
+        if (this._initialized) await this._mongoClient.close();
+    }
+
+    private async _saveToGlobalConfigSets(globalConfigSet : GlobalConfigurationSet):Promise<void>{
+        try {
+            const result = await this._collectionGlobalConfigSets.insertOne({
+                    schemaVersion: globalConfigSet.schemaVersion,
+                    iterationNumber: globalConfigSet.iterationNumber,
+                    parameters: globalConfigSet.parameters,
+                    featureFlags: globalConfigSet.featureFlags,
+                    secrets: globalConfigSet.secrets
+                }
+            );
```

Comment:
Please remove const result variable because it will not be used. Using await is enough.